content/fast/defer-unused-css/index.md

```diff
@@ -0,0 +1,150 @@
+---
+page_type: guide
+title: Defer unused CSS
+author: demianrenzulli
+description: |
+  Learn how to defer unused CSS with the goal of optimizing the Critical Rendering Path, and improve FCP (First Contentful Paint).
+web_lighthouse:
+- unused-css-rules
+- render-blocking-resources
+web_updated_on: 2019-02-17
+web_published_on: 2019-02-17
+wf_blink_components: N/A
+---
+
+# Defer unused CSS
+
+CSS files are [render-blocking resources](https://developers.google.com/web/tools/lighthouse/audits/blocking-resources), meaning that they have to be loaded and processed before the browser can start rendering the page. For this reason, web pages that contain unnecessarily large or unused styles may take longer to render, increasing the likelihood that a user may bounce: 53% of mobile visits bounce if the page does not load [in 3 seconds or less](https://www.thinkwithgoogle.com/intl/en-154/insights-inspiration/research-data/need-mobile-speed-how-mobile-latency-impacts-publisher-revenue/).
+
+In this guide, you’ll learn how to defer unused CSS with the goal of optimizing the [Critical Rendering Path](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/), and improving [FCP (First Contentful Paint)](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint).
+
+## Loading CSS in a suboptimal way
+
+The following example contains an accordion with three hidden paragraphs of text, each of which is styled with a different class.
+
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
+  <iframe
+    allow="geolocation; microphone; camera; midi; encrypted-media"
+    src="https://glitch.com/embed/#!/embed/defer-css-unoptimized?path=index.html&previewSize=100&attributionHidden=true"
+    alt="defer-css-unoptimized on Glitch"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+This page requests a CSS file with eight classes, but not all of them are necessary to render the "visible" content. 
+
+The goal of this guide is to optimize this page, so only the **critical** styles are loaded synchronously, while the rest (like the ones applied to paragraphs), are loaded in a non-blocking way.
+
+## Measure
+
+Run [Lighthouse](https://web.dev/fast/discover-performance-opportunities-with-lighthouse/#run-lighthouse-from-chrome-devtools) on [the page](https://defer-css-unoptimized.glitch.me/) and go to the **Performance** section.
+
+The report shows the **FCP** metric with a value of "1s", and the opportunity **Eliminate render-blocking resources**, pointing to the **style.css** file:
```

Comment:
IMPORTANT: I ran lighthouse on the page-- and I don't see the opportunity for deferring unused css-- it is showing up in passed audits.

Reply:
Thanks Meggin.

Regarding LH: It seems like, being this a relatively small CSS file, LH shows "Defer unused CSS" as "Passed", while "Eliminate render-blocking resources" it's still being shown as opportunity.
For that reason, I used the latter show the impact before/after implementing this optimization, but it's true that's not "exactly" the suggestion we are trying to demonstrate.

Some ideas to improve this:

- Adding a note explaining that "Defer unused CSS" doesn't show up, because it's a small file.
- Removing the LH opportunity improvement, to avoid confusions, and focusing only on metrics improvements.
- Modifying the demo (maybe using a larger CSS), to make the LH test fail first. 

I personally like the last one, but wouldn't like to end up having a demo that's so large that it ends up making it less readable for developers.
I saw this same problem when working on other demos as well.

Any thoughts are welcomed.

Reply:
So focusing on the opportunity to eliminate render-blocking resources might work. Maybe try that?

Reply:
Exactly. 
So, at the beginning of the article, I'm using the "Eliminate render-blocking resources" Opportunity, which appears when running LH on the non-optimized page:

https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://defer-css-unoptimized.glitch.me/

After applying the optimization, that opportunity moves to "Passed Audits" (and the whole "Opportunities" section disappear, as you can see here:

https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://defer-css-optimized.glitch.me/

If you think that's ok, now that it's clear why I didn't use the "Defer unused CSS" one, we can leave it in that way.